server/src/test/java/org/elasticsearch/cluster/service/TaskBatcherTests.java

```diff
@@ -271,7 +271,9 @@ public void processed(String source) {
                 public void onFailure(String source, Exception e) {
                     fail(ExceptionsHelper.detailedMessage(e));
                 }
-            })) ;
+            })) {
```

Comment:
What is the intention of this loop? It seems sketchy. It repeatedly puts entries into the `tasks` Map until the random int key that is chosen is duplicated (that is when put will return non-null, the already existing task entry). Perhaps this could be simplified to not need a loop like this?

Reply:
> What is the intention of this loop?

This also looked strange to me but I didn't want to change it since I have no good knowleadge of the test and I wanted to keep this PR strictly on clean-ups.
I agree on your reading of what the loop does and will change it so it doesn't need an empty loop body anymore. It's confusing.